src/pages/dashboard/DashboardContext.tsx

```diff
@@ -0,0 +1,49 @@
+import { FC, PropsWithChildren, createContext, useState, useEffect } from 'react'
+
+export interface DashboardContextType {
+  setAllLinesChartsIsLoading: React.Dispatch<React.SetStateAction<boolean>>
+  setWorstLineIsLoading: React.Dispatch<React.SetStateAction<boolean>>
+  setDayTimeChartIsLoading: React.Dispatch<React.SetStateAction<boolean>>
+  setAllLinesChartsIsEmpty: React.Dispatch<React.SetStateAction<boolean>>
+  setWorstLineIsEmpty: React.Dispatch<React.SetStateAction<boolean>>
+  setDayTimeChartIsEmpty: React.Dispatch<React.SetStateAction<boolean>>
+  isDataLoading: boolean
+  isDataEmpty: boolean
+}
+export const DashboardCtx = createContext<DashboardContextType>({} as DashboardContextType)
+export const DashboardContext: FC<PropsWithChildren> = ({ children }) => {
+  // loading
+  const [allLinesChartsIsLoading, setAllLinesChartsIsLoading] = useState(false)
+  const [worstLineIsLoading, setWorstLineIsLoading] = useState(false)
+  const [allDayTimeChartIsLoading, setDayTimeChartIsLoading] = useState(false)
+  // is empty
+  const [allLineChartsIsEmpty, setAllLinesChartsIsEmpty] = useState(false)
+  const [worstLineIsEmpty, setWorstLineIsEmpty] = useState(false)
+  const [dayTimeChartIsEmpty, setDayTimeChartIsEmpty] = useState(false)
+  const [isDataEmpty, setIsDataEmpty] = useState(false)
+  const [isDataLoading, setIsDataLoading] = useState(false)
+
+  useEffect(() => {
+    setIsDataEmpty(allLineChartsIsEmpty && worstLineIsEmpty && dayTimeChartIsEmpty)
+  }, [allLineChartsIsEmpty, worstLineIsEmpty, dayTimeChartIsEmpty])
+
+  useEffect(() => {
+    setIsDataLoading(allLinesChartsIsLoading || worstLineIsLoading || allDayTimeChartIsLoading)
+  }, [allLinesChartsIsLoading, worstLineIsLoading, allDayTimeChartIsLoading])
+
+  return (
+    <DashboardCtx.Provider
+      value={{
+        setAllLinesChartsIsLoading,
+        setWorstLineIsLoading,
+        setDayTimeChartIsLoading,
+        setAllLinesChartsIsEmpty,
+        setWorstLineIsEmpty,
+        setDayTimeChartIsEmpty,
+        isDataLoading,
+        isDataEmpty,
```

Comment:
When someone consumes this context, I don't want them to make guesses on whether there's an "and" (`&&`) or "or" (`||`) condition between the different statuses. I suggest that we could try a naming convention that corresponds with the [Array.prototype.every()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every) and [Array.prototype.some()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some)

```suggestion
        isSomeDataLoading,
        isAllDataEmpty,
```